enterprise/frontend/src/metabase-enterprise/rich_text_editing/tiptap/extensions/LinkHoverMenu/LinkHoverMenu.tsx

```diff
@@ -0,0 +1,141 @@
+import type { Editor } from "@tiptap/core";
+import {
+  useCallback,
+  useEffect,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from "react";
+
+import { Ellipsified } from "metabase/common/components/Ellipsified";
+import ExternalLink from "metabase/common/components/ExternalLink";
+import { ActionIcon, Box, Card, FixedSizeIcon, Flex } from "metabase/ui";
+
+import S from "../PlainLink/PlainLink.module.css";
+
+const HOVER_TIMEOUT_MS = 150;
+const MAX_W = 480;
+
+interface LinkHoverMenuProps {
+  editor: Editor;
+  editable: boolean;
+}
+
+export const LinkHoverMenu = ({ editor, editable }: LinkHoverMenuProps) => {
+  const [hoverPosition, setHoverPosition] = useState({ top: 0, left: 0 });
+  const [hoveredLink, setHoveredLink] = useState<HTMLElement | null>(null);
+  const hoverTimeoutRef = useRef<ReturnType<typeof setTimeout> | null>(null);
+
+  const clearHoverTimeout = useCallback(() => {
+    if (hoverTimeoutRef.current) {
+      clearTimeout(hoverTimeoutRef.current);
+      hoverTimeoutRef.current = null;
+    }
+  }, []);
+
+  const startHoverTimeout = useCallback(() => {
+    clearHoverTimeout();
+    hoverTimeoutRef.current = setTimeout(() => {
+      setHoveredLink(null);
+    }, HOVER_TIMEOUT_MS);
```

Comment:
the hover timeout is a bit inconsistent. If I move my mouse slowly, it's possible for the popover to completely disappear :/